configs/defs/generic_linux.textproto

```diff
@@ -9558,3 +9558,931 @@ benchmark_configs: {
+      "This configuration ensures that audit events for processes that start prior to auditd are captured, enabling early detection of potential malicious activity."
+    rationale: 
+      "Auditing processes that start prior to auditd ensures comprehensive monitoring of all system activity, reducing the risk of undetected malicious activity."
+    remediation: 
+      "Edit `/etc/default/grub` to include `audit=1` in the `GRUB_CMDLINE_LINUX` configuration and update the GRUB configuration with the command `update-grub`."
+    cis_benchmark: {
+      profile_level: 2
+      severity: HIGH
+    }
+    scan_instructions:
+      "scan_type_specific:{instance_scanning:{"
+      "  check_alternatives:{"
+      "    file_checks:{"
+      "      files_to_check:{single_file:{path:\"/boot/grub/grub.cfg\"}}"
+      "      content_entry:{"
+      "        match_type: ALL_MATCH_ANY_ORDER"
+      "        match_criteria: {"
+      "          filter_regex: \"\\\\s*linux.*\\\\s+audit=1\\b.*\""
+      "          expected_regex: \"\\\\s*linux.*\\\\s+audit=1\\b.*\""
+      "        }"
+      "      }"
+      "    }"
+      "  }"
+      "}}"
+  }
+}
+
+
+
+benchmark_configs: {
+  id: "auditd-installed"
+  compliance_note: {
+    title: "Ensure auditd is installed"
+    description:
+      "auditd is the userspace component to the Linux Auditing System. "
+      "It's responsible for writing audit records to the disk"
+    rationale:
+      "The capturing of system events provides system administrators with "
+      "information to allow them to determine if unauthorized access to their system is occurring."
+    remediation:
+      "Run the following command and verify auditd is installed: \n"
+      "```# dpkg-query -s auditd &>/dev/null && echo auditd is installed\n```"
+      "Run the following command to verify audispd-plugins is installed:\n"
+      "```# dpkg-query -s audispd-plugins &>/dev/null && echo audispd-plugins is installed\n```"
+      "Run the following command to Install auditd and audispd-plugins\n"
+      "```# apt install auditd audispd-plugins\n```"
+    cis_benchmark: {
+      profile_level: 2
+      severity: LOW
+    }
+    scan_instructions:
+      "generic:{check_alternatives:{"
+      "  file_checks:{"
+      "    files_to_check:{single_file:{path:\"/usr/bin/auditd\"}}"
+      "    existence:{should_exist: true}"
+      "  }"
+      "}"
+      "check_alternatives:{"
+      "  file_checks:{"
+      "    files_to_check:{single_file:{path:\"/usr/sbin/auditd\"}}"
+      "    existence:{should_exist: true}"
+      "  }"
+      "}"
+      "check_alternatives:{"
+      "  file_checks:{"
+      "    files_to_check:{single_file:{path:\"/bin/auditd\"}}"
+      "    existence:{should_exist: true}"
+      "  }"
+      "}"
+      "check_alternatives:{"
+      "  file_checks:{"
+      "    files_to_check:{single_file:{path:\"/sbin/auditd\"}}"
+      "    existence:{should_exist: true}"
+      "  }"
+      "}"
+      "check_alternatives:{"
```

Comment:
We only need the check for /bin not existing for fallback benchmarks that we're running on binary-only containers that might not have a filesystem. The generic_linux benchmarks won't be running on containers like that so there's no need to check for this.